servers/src/main/java/tachyon/worker/file/FileDataManager.java

```diff
@@ -116,7 +175,10 @@ public synchronized void persistFile(long fileId, List<Long> blockIds) throws IO
     outputStream.flush();
     outputChannel.close();
     outputStream.close();
-    mPersistedFiles.add(fileId);
+    synchronized (this) {
```

Comment:
but that means that if you sometimes use `synchronized(this)` and other times `synchronized(mPersistedFiles)` the two synchronized blocks will actually not be synchronized, correct?